packages/wrangler/src/__tests__/logout.test.ts

```diff
@@ -0,0 +1,54 @@
+import { mkdirSync, existsSync } from "fs";
+import { runWrangler } from "./run-wrangler";
+import { runInTempDir } from "./run-in-tmp";
+import os from "node:os";
+import path from "node:path";
+import { initialise } from "../user";
+import { writeUserConfig } from "./whoami.test";
+
+const ORIGINAL_CF_API_TOKEN = process.env.CF_API_TOKEN;
+const ORIGINAL_CF_ACCOUNT_ID = process.env.CF_ACCOUNT_ID;
+
+describe("wrangler", () => {
+  runInTempDir();
+
+  beforeEach(() => {
+    // Override where the home directory is so that we can specify a user config
+    mkdirSync("./home");
+    jest.spyOn(os, "homedir").mockReturnValue("./home");
+
+    delete process.env.CF_API_TOKEN;
+    delete process.env.CF_ACCOUNT_ID;
+  });
+
+  afterEach(() => {
+    // Reset any changes to the environment variables
+    process.env.CF_API_TOKEN = ORIGINAL_CF_API_TOKEN;
+    process.env.CF_ACCOUNT_ID = ORIGINAL_CF_ACCOUNT_ID;
+  });
+
+  describe("logout", () => {
+    it("should exit with a message stating the user is not logged in", async () => {
+      await initialise();
+      const { stdout } = await runWrangler("logout");
+      expect(stdout).toMatchInlineSnapshot(`"Not logged in, exiting..."`);
+    });
+
+    it("should logout user that has been properly loggedin", async () => {
+      writeUserConfig("some-oauth-tok", "some-refresh-tok");
+
+      await initialise();
+      const { stdout } = await runWrangler("logout");
```

Comment:
Logout makes a fetch request to the CF OAuth revoke api, not sure if/how to test for this request being successful

Reply:
We need to mock out the raw `node-fetch` module since logout (and other things) hit this directly, rather than going through the cfetch API.

Reply:
I'll see if I can solve that tonight or tomorrow morning so we can land this safely.